google/resource_compute_instance.go

```diff
@@ -1291,108 +1286,57 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 		}
 	}
 
-	networkInterfacesCount := d.Get("network_interface.#").(int)
+	networkInterfaces, err := expandNetworkInterfaces(d, config)
+	if err != nil {
+		return fmt.Errorf("Error getting network interface from config: %s", err)
+	}
+
 	// Sanity check
-	if networkInterfacesCount != len(instance.NetworkInterfaces) {
+	if len(networkInterfaces) != len(instance.NetworkInterfaces) {
 		return fmt.Errorf("Instance had unexpected number of network interfaces: %d", len(instance.NetworkInterfaces))
 	}
-	for i := 0; i < networkInterfacesCount; i++ {
+
+	var updatesToNIWhileStopped []func() *compute.Operation, error
+	for i := 0; i < len(networkInterfaces); i++ {
 		prefix := fmt.Sprintf("network_interface.%d", i)
+		networkInterface := networkInterfaces[i]
 		instNetworkInterface := instance.NetworkInterfaces[i]
-		networkName := d.Get(prefix + ".name").(string)
+
+		networkName := networkInterface.Name
+		network := networkInterface.Network
+		subnetwork := networkInterface.Subnetwork
+		updateDurringStop := d.HasChange(prefix+".subnetwork") || d.HasChange(prefix+".network") || d.HasChange(prefix+".subnetwork_project") || d.HasChange(prefix+".network_ip")
 
 		// Sanity check
 		if networkName != instNetworkInterface.Name {
 			return fmt.Errorf("Instance networkInterface had unexpected name: %s", instNetworkInterface.Name)
 		}
 
-		if d.HasChange(prefix + ".access_config") {
-
-			// TODO: This code deletes then recreates accessConfigs.  This is bad because it may
-			// leave the machine inaccessible from either ip if the creation part fails (network
-			// timeout etc).  However right now there is a GCE limit of 1 accessConfig so it is
-			// the only way to do it.  In future this should be revised to only change what is
-			// necessary, and also add before removing.
-
-			// Delete any accessConfig that currently exists in instNetworkInterface
-			for _, ac := range instNetworkInterface.AccessConfigs {
-				op, err := config.clientCompute.Instances.DeleteAccessConfig(
-					project, zone, instance.Name, ac.Name, networkName).Do()
+		if d.HasChange(prefix + ".subnetwork") {
+			if !d.HasChange(prefix + ".network") {
+				subnetProjectField := d.Get(prefix + ".subnetwork_project").(string)
```

Comment:
I think we just want to send in the `prefix + .subnetwork_project` string? This sends the value in where I thought the path to the field was expected.

Reply:
nice catch! that's correct